data-prepper-plugins/saas-source-plugins/crowdstrike-source/src/main/java/org/opensearch/dataprepper/plugins/source/crowdstrike/CrowdStrikeService.java

```diff
@@ -0,0 +1,89 @@
+package org.opensearch.dataprepper.plugins.source.crowdstrike;
+import io.micrometer.core.instrument.Timer;
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.opensearch.dataprepper.plugins.source.crowdstrike.models.CrowdStrikeApiResponse;
+import org.opensearch.dataprepper.plugins.source.crowdstrike.models.CrowdStrikeIndicatorResult;
+import org.opensearch.dataprepper.plugins.source.crowdstrike.rest.CrowdStrikeRestClient;
+import org.opensearch.dataprepper.plugins.source.crowdstrike.utils.CrowdStrikeNextLinkValidator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.util.UriComponentsBuilder;
+import javax.inject.Named;
+import java.net.URI;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import static org.opensearch.dataprepper.plugins.source.crowdstrike.utils.Constants.BATCH_SIZE;
+import static org.opensearch.dataprepper.plugins.source.crowdstrike.utils.Constants.FILTER_KEY;
+import static org.opensearch.dataprepper.plugins.source.crowdstrike.utils.Constants.LAST_UPDATED;
+import static org.opensearch.dataprepper.plugins.source.crowdstrike.utils.Constants.LIMIT_KEY;
+
+/**
+ * This service manages the interaction with CrowdStrike's API endpoints, handles pagination
+ * and provides methods for fetching indicators from CrowdStrike.
+ */
+@Named
+public class CrowdStrikeService {
+
+    private final CrowdStrikeRestClient crowdStrikeRestClient;
+    private static final Logger log = LoggerFactory.getLogger(CrowdStrikeService.class);
+    private static final String BASE_URL = "https://api.crowdstrike.com/";
+    private static final String COMBINED_URL = "https://api.crowdstrike.com/intel/combined/indicators/v1";
+    private final Timer searchCallLatencyTimer;
+
+
+    public CrowdStrikeService(CrowdStrikeRestClient crowdStrikeRestClient, PluginMetrics pluginMetrics) {
+        this.crowdStrikeRestClient = crowdStrikeRestClient;
+        this.searchCallLatencyTimer = pluginMetrics.timer("searchCallLatencyTimer");
+    }
+
+    /**
+     * Retrieves all indicator data from CrowdStrike in a paginated fashion.
+     *   @param startTime       The start timestamp (inclusive) for filtering indicators.
+     *   @param endTime         The end timestamp (exclusive) for filtering indicators.
+     *   @param paginationLink  An optional pagination URL suffix (used when fetching next pages).
+     *   @return                A {@link CrowdStrikeApiResponse} containing response body and headers.
+     */
+    public CrowdStrikeApiResponse getAllContent(Long startTime, Long endTime, String paginationLink) {
```

Comment:
minor: can we rename method and append to code comments to indicate this indicator API if it's not a generic get for all api responses